### PR TITLE
Bug 1136535 - webapp object should be JSON.stringifiedable 

### PR DESCRIPTION
**Pull request by cctuan**
https://bugzilla.mozilla.org/show_bug.cgi?id=1136535

### CODE REVIEW

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=49ffe7e09af2)

File: build/copy-build-stage-data.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ function moveExternalApp(webapp, source, destination) {
     appPackage.copyTo(destination, 'application.zip');
     updateManifest.copyTo(destination, 'update.webapp');
   } else {
-    webapp.manifestFile.copyTo(destination, 'manifest.webapp');
+    var manifestFile = utils.getFile( webapp.manifestFilePath);
```
Comment (RickyChien) [inline build/copy-build-stage-data.js]: ```utils.getFile( webapp.manifestFilePath);```
                      ^
remove this whitespace

File: build/rebuild.js [hunk 2]
```diff
@@ -127,7 +127,7 @@ exports.execute = function(options) {
 
         if (buildFile.exists() ||
             (utils.isExternalApp(webapp) && !webapp.pckManifest) ||
-            !webapp.profileDirectoryFile.exists()) {
+            !utils.getFile(webapp.profileDirectoryFilePath).exists()) {
```
Comment (RickyChien) [inline build/rebuild.js]: Use utils.fileExists(webapp.profileDirectoryFilePath) instead.

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=6ea5f6c172d4)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=82f8a0198d91)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=15181777b534)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8ee828f3c16b)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=005c065bac08)

**Pull request merged**